internal/wasm/module_context.go

```diff
@@ -105,13 +109,31 @@ func (f *FunctionInstance) ResultTypes() []publicwasm.ValueType {
 }
 
 // Call implements wasm.Function Call
-func (f *FunctionInstance) Call(ctx publicwasm.Module, params ...uint64) ([]uint64, error) {
+func (f *FunctionInstance) Call(m publicwasm.Module, params ...uint64) (ret []uint64, err error) {
+	if err = f.failIfClosed(); err != nil {
```

Comment:
this guard is new. before we still allowed new calls to modules that were already closed (ex if you kept a ref to the module).